### ISSUE TITLE
Issue with "storage.StorageArea": onChanged is not available in Firefox, but is in Chromium-based browsers

### ISSUE THREAD

**Issue by BigBlueHat** [mdn/content#11411]
MDN URL: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea

#### What information was incorrect, unhelpful, or incomplete?
The `storage.StorageArea` page claims "Full Support" for Firefox, but Firefox currently lacks the `onChanged` object as a member of the `StorageArea` objects found at `browser.storage.local` or `managed` or `sync`.

#### Specific section or headline?
The `onChanged` object should appear in the list of supported/valid methods for `StorageArea`, and Firefox's lack of support for that feature should be noted.

#### What did you expect to see?
Clarity about what is and is not supported by Firefox vs. other browsers.

#### Did you test this? If so, how?
I shipped an extension--which worked fine in Chrome, Edge, etc.--using the `chrome.storage.local.onChanged.addListener()` method which I (quickly) learned failed in Firefox. Elsewhere in the extension I used the `chrome.storage.onChanged.addListener()` method, so I had overlooked this difference which was unique to Firefox.

<details>
<summary>MDN Content page report details</summary>

* Folder: `en-us/mozilla/add-ons/webextensions/api/storage/storagearea`
* MDN URL: https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/StorageArea
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/mozilla/add-ons/webextensions/api/storage/storagearea/index.md
* Last commit: https://github.com/mdn/content/commit/629960fa07e15f2354d75daf14f149d4942539ec
* Document last modified: 2021-10-26T21:28:18.000Z

</details>

**Comment by odebroqueville** [comment]
The information on the following page states that onChanged is compatible with Firefox 45+ :
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/storage/onChanged

However, I've tested the following code in the content script of one of my Firefox web extensions and I got the result \<unavailable\> when console logging Object.keys(changes):

```
// Storage change event listener
browser.storage.onChanged.addListener(handleStorageChange);

async function handleStorageChange(changes, area) {
	let oldSearchEngines = JSON.parse(JSON.stringify(searchEngines));
	let ids = Object.keys(changes);
	let data;
	if (logToConsole) {
		console.log('The following changes have occured:\n');
		console.log(ids);
	}
	switch (area) {
		case 'local':
            ...
			break;
		case 'sync':
            ...
			break;
		default:
			break;
	}
}
```

**Comment by Rob--W** [comment]
Thanks for the report. The StorageArea.onChanged event was introduced later.  "Chrome/Edge" are built from a shared code base, so them supporting a feature does not automatically mean that a feature is standardized. Notable independent implementations of the WebExtensions API are Safari and Firefox.

But given the reasonable expectation of interoperability of this event, along with valid reasons for needing this event at all, I have filed a feature request on Firefox's issue tracker at https://bugzilla.mozilla.org/show_bug.cgi?id=1758475

The documentation should be improved as follows:

- add BCD entry for StorageArea.onChanged
- add documentation entry for StorageArea.onChanged, which should also refer to the top-level storage.onChanged event as a fallback when a method is not implemented.

**Comment by mhammond** [comment]
FWIW, desktop's `storage.sync` area supports this.

**Comment by Rob--W** [comment]
Support for `StorageArea.onChanged` has landed in Firefox 101. We still have to update the documentation (mdn/content and mdn though, per https://bugzilla.mozilla.org/show_bug.cgi?id=1758475#c5.

An article for `StorageArea.onChanged` (i.e. `storage.local.onChanged`, `storage.sync.onChanged` and `storage.managed.onChanged`) needs to be added. It is largely the same as the existing `storage.onChanged` event, except the latter receives a second parameter that shows which area has been updated ("local", "managed" or "sync").

**Comment by rebloor** [comment]
Resolved by [Bug 1758475](https://bugzilla.mozilla.org/show_bug.cgi?id=1758475) Support StorageArea.onChanged (onChanged on storage.local/sync/managed) and associated documentation changes in [Support for StorageArea.onChanged and Session #16450](https://github.com/mdn/browser-compat-data/pull/16450) (BCD) and https://github.com/mdn/content/pull/16873 (docs)
